compiler/rustc_borrowck/src/diagnostics/explain_borrow.rs

```diff
@@ -359,19 +359,38 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         let borrow_region_vid = borrow.region;
         debug!(?borrow_region_vid);
 
-        let region_sub = self.regioncx.find_sub_region_live_at(borrow_region_vid, location);
+        let mut region_sub = self.regioncx.find_sub_region_live_at(borrow_region_vid, location);
         debug!(?region_sub);
 
-        match find_use::find(body, regioncx, tcx, region_sub, location) {
+        let mut use_location = location;
+        let mut use_in_later_iteration_of_loop = false;
+
+        if region_sub == borrow_region_vid {
+            // when region_sub is the same as borrow_region_vid, which means that the location where
+            // borrow issued is the same location that invalidates the reference, and this likely
+            // heppens in a loop iteration. Using the loop terminator location instead for 'location'
+            // to for find_sub_region_live_at(..)
```

Comment:
```suggestion
            // When `region_sub` is the same as `borrow_region_vid` (the location where
            // the borrow is issued is the same location that invalidates the reference),
            // this is likely a loop iteration - in this case, try using the loop 
            // terminator location in `find_sub_region_live_at`.
```

You might need to fix the line wrapping.

Reply:
Modified as suggested.